src/http/h1/encode.rs

```diff
@@ -107,50 +108,54 @@ impl Chunked {
             Chunked::End => return Ok(0),
             _ => {}
         }
+
         let mut n = {
-            let pieces = match *self {
+            // TODO: Find a better way to concatente
+            //    Could use write! if we reserved the required capacity upfront
+            match *self {
```

Comment:
Ok, I wont lie your statements are a bit above my pay grade. Mostly I was just trying to get rid of the allocations that come from `.concat()`, but can deffer that for now.

Do you mind if I just inline `AtomicWrite` since its just a simple passthrough, or would you rather keep it to be able to extend later?